pydm/widgets/base.py

```diff
@@ -273,8 +274,7 @@ def generate_context_menu(self):
             menu = QMenu(parent=self)
 
         kwargs = {'channels': self.channels_for_tools(), 'sender': self}
-        if hasattr(self.app, 'assemble_tools_menu'):
-            self.app.assemble_tools_menu(menu, widget_only=True, **kwargs)
+        tools.assemble_tools_menu(menu, widget_only=True, **kwargs)
```

Comment:
I can’t test for myself right now - does this screw up the Designer right-click menu?